### PR TITLE
added RawAttributes interface

### LINKED ISSUE #161: Custom Token Factory and AllClaims Interface
The newer versions of bascule no longer have the `FullView()` function in the [`Attributes` interface](https://github.com/xmidt-org/bascule/blob/main/token.go#L39), talaria needs a custom implementation of `Attributes` that also implements a talaria specific interface:
```
type AllClaimer interface{
  GetAllClaims() map[string]interface{}
}
```
(Open to a different interface/function name, but you get the idea)

Then [this middleware](https://github.com/xmidt-org/talaria/blob/main/middleware.go#L18) can do a type assertion on the created token's `Attributes` to an `AllClaimer` in order to get the claims needed for the device metadata.

In order to get its own implementation into the token in the context, talaria should have its own token factory, similar to this one in bascule:
https://github.com/xmidt-org/bascule/blob/main/basculehttp/tokenFactory.go#L128
The token factory should create the talaria attribute implementation and put it in the token, instead of using `bascule.NewAttributes()`.

Another example of a custom token factory is here (wip):
https://github.com/xmidt-org/argus/blob/9f7c5a2e7fc2ed9c733783c8311f2330284ee14d/tokenFactory.go#L38

Tasks I know of:
- [ ] Create `AllClaimer` type interface.
- [ ] Create a struct that implements the bascule `Attributes` interface and `AllClaimer`.
- [ ] Create a new token factory that builds a bearer token using the new struct.
- [ ] Do a type assertion on the attributes from the token in the context in order to get all the claims and add them to the device metadata.

### PR DESCRIPTION
**Pull request by j-mai**
This PR allows for talaria to get all of a JWT's claims from an Attribute.

Fixes #161

### CODE REVIEW

File: middleware.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func DeviceMetadataMiddleware(delegate http.Handler) http.Handler {
 			metadata.SetSessionID(ksuid.New().String())
 
 			if auth, ok := bascule.FromContext(ctx); ok {
-				metadata.SetClaims(auth.Token.Attributes().FullView())
+				metadata.SetClaims(auth.Token.Attributes().(rawAttributes).GetRawAttributes())
```
Comment (kcajmagic) [inline middleware.go:26]: We should add a check to see if it is a `RawAttributes`

```suggestion
				if tokenAttributes, ok := auth.Token.Attributes().(RawAttributes); ok {
					metadata.SetClaims(tokenAttributes.GetRawAttributes())
				}
```

File: mocks_test.go [hunk 2]
```diff
@@ -65,3 +66,13 @@ func (m *mockDeviceAccess) authorizeWRP(ctx context.Context, message *wrp.Messag
 	arguments := m.Called(ctx, message)
 	return arguments.Error(0)
 }
+
+type mockJWTParser struct {
+	mock.Mock
+}
+
+func (parser *mockJWTParser) ParseJWT(token string, claims jwt.Claims, parseFunc jwt.Keyfunc) (*jwt.Token, error) {
+	arguments := parser.Called(token)
```
Comment (kcajmagic) [inline mocks_test.go:75]: all parameters should be passed in mock struct
```suggestion
	arguments := parser.Called(token, claims, parseFunc)
```

File: tokenFactory_test.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+	"github.com/dgrijalva/jwt-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/xmidt-org/bascule"
+	"github.com/xmidt-org/bascule/basculehttp"
+	"github.com/xmidt-org/bascule/key"
+)
+
+func TestRawAttributesBearerTokenFactory(t *testing.T) {
+
```
Comment (kcajmagic) [inline tokenFactory_test.go:18]:
```suggestion
```

File: tokenFactory_test.go [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+			token, err := f.ParseAndValidate(context.Background(), req, "", tc.value)
+
+			assert.Equal(tc.expectedToken, token)
+
+			if tc.expectedErr == nil || err == nil {
+				assert.Equal(tc.expectedErr, err)
+			} else {
+				assert.Contains(err.Error(), tc.expectedErr.Error())
+			}
+
```
Comment (kcajmagic) [inline tokenFactory_test.go:144]:
```suggestion
```

File: tokenFactory_test.go [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+		})
+	}
+
```
Comment (kcajmagic) [inline tokenFactory_test.go:147]:
```suggestion
```

**Review by kcajmagic — CHANGES_REQUESTED** [review]
looks good, just a couple of minor changes.

**Review by kcajmagic — APPROVED** [review]
Awesome, looks good.

**Review by kristinaspring — COMMENTED** [review]
Just one question - not sure about the desired middleware functionality.

**Review by kristinaspring — APPROVED** [review]
looks good 💯

**Pull request merged**